S3/ConnMan.py

```diff
@@ -108,20 +116,21 @@ def match_hostname(self):
         except ValueError: # empty SSL cert means underlying SSL library didn't validate it, we don't either.
             return
         except ssl.CertificateError, e:
-            self.match_hostname_aws(cert, e)
+            self.forgive_wildcard_cert(cert, e)
 
     @staticmethod
     def _https_connection(hostname, port=None):
         check_hostname = True
         try:
             context = http_connection._ssl_context()
-            # S3's wildcart certificate doesn't work with DNS-style named buckets.
-            if (hostname.endswith('.amazonaws.com') or hostname.endswith('.amazonaws.com.cn')):
+            # Wilcard certificates do not work with DNS-style named buckets.
+            bucket_name, success = getBucketFromHostname(hostname)
+            if ('.' in bucket_name and success):
```

Comment:
[cosmetic] As it's python, no need for global parenthesis around the if statement ;-)
+ [optimization] the "success" check should probably come first as it's faster to check.